### ISSUE TITLE
c++abi not linked

### ISSUE THREAD

**Issue by suriya1611** [theodelrieu/conan-android-ndk#2]
I used android-ndk package to compile ICU library. Cross compiling from Mac OSX to android-armv7.
I get the following  undefined references.
```
makeconv.o: In function `__clang_call_terminate':
makeconv.cpp:(.text.__clang_call_terminate[__clang_call_terminate]+0x4): undefined reference to `__cxa_begin_catch'
makeconv.cpp:(.text.__clang_call_terminate[__clang_call_terminate]+0x8): undefined reference to `std::terminate()'
makeconv.o:(.data.DW.ref.__gxx_personality_v0[DW.ref.__gxx_personality_v0]+0x0): undefined reference to `__gxx_personality_v0'

../../lib/libicuuc.a(putil.ao): In function `putil_cleanup()':
undefined reference to `operator delete(void*)'
undefined reference to `operator delete(void*)'
```

**Comment by theodelrieu** [comment]
Can you paste the linker command line? If you can also provide a small reproducable example that would be great (though the command line should be enough)

**Comment by suriya1611** [comment]
hi i checked the config.log of icu, c++abi is being linked. When i checked the log file, there were errors saying "cannot find lpthreads", and many of C libraries missing. Should i link libc and check if it works?

I guess in sysroot folder, include path specified for asm files, but not specified for other C header files. Am i right about that? 
"sysroot/usr/include/arm-linux-androideabi" is added in compiler flags. 
Should "sysroot/usr/include" also be added?

**Comment by theodelrieu** [comment]
>Should "sysroot/usr/include" also be added

No, this would break stuff related to include order (there are some `#include_next` directives in the STL, you do not want to make those fail). Also it is already handled by the `--sysroot` flag.

Could you upload the full log file? If you have a conan recipe for ICU that I can test, it will be easier to debug.

**Comment by suriya1611** [comment]
[config.log](https://github.com/theodelrieu/conan-android-ndk/files/2773994/config.log)
https://github.com/suriya1611/icu-conan/blob/master/conanfile.py
https://github.com/suriya1611/conan-icu-cross-build

I have a icu-cross-build package from which i copy the cross build directory for cross compiling - based on os_build and arch_build.

This conan recipe was taken from bincrafters repo. Since they didnt have cross compiling in it. I had to edit it.
reference : https://github.com/bincrafters/conan-icu
in the configure method, they set the libcxx value as "libstdc++11". Will that work on android-ndk?

Btw im building from Mac OSX to android.

**Comment by suriya1611** [comment]
is there anything wrong with the cross compilation?

**Comment by theodelrieu** [comment]
`os_build` and `arch_build` should never be used for regular libraries, only for build tools or cross-building toolchains.

I will try the original recipe on my macOS this morning.

EDIT: I've explained cross-building with Conan in more detail in this [talk](https://www.youtube.com/watch?v=Gm2h8ZWCEH4)

**Comment by suriya1611** [comment]
Yeah i should've deleted that setting in the regular library, but it needs to be there for cross compiling package right? I mean the one which contains the build folder for the ```os_build``` and ```os_arch```.

**Comment by theodelrieu** [comment]
Well, it seems you have to have them in the case of ICU, because for some reason cross-building it requires building it for the build machine first. I've never seen that before...

I'll try to make it work, but this seems to be a recipe issue, have you opened one in conan-community?

**Comment by suriya1611** [comment]
Recipe was modified by me to support cross compilation, the original doesn't seem to support it. 
Anyways i will open an issue for cross compilation in conan community.

**Comment by theodelrieu** [comment]
Ok, making it work will take some time, since you have to trigger the build twice, once with `os_build`/`arch_build`, once with `os`/`arch`.

There is some refactoring to be done in this recipe, I will post some remarks once you open the issue in community :)

**Comment by theodelrieu** [comment]
Any news on this?
